bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java

```diff
@@ -625,8 +626,12 @@ public Journal(File journalDirectory, ServerConfiguration conf, LedgerDirsManage
         this.maxGroupWaitInNanos = TimeUnit.MILLISECONDS.toNanos(conf.getJournalMaxGroupWaitMSec());
         this.bufferedWritesThreshold = conf.getJournalBufferedWritesThreshold();
         this.bufferedEntriesThreshold = conf.getJournalBufferedEntriesThreshold();
-        this.cbThreadPool = Executors.newFixedThreadPool(conf.getNumJournalCallbackThreads(),
-                                                         new DaemonThreadFactory());
+        if (conf.getNumJournalCallbackThreads() > 0) {
+            this.cbThreadPool = Executors.newFixedThreadPool(conf.getNumJournalCallbackThreads(),
+                                                         new DefaultThreadFactory("bookie-journal-callback"));
+        } else {
+            this.cbThreadPool = MoreExecutors.newDirectExecutorService();
```

Comment:
this thread pool gets used on two different threads(JournalThread & ForceWriteThread). Are you sure this is safe?

Reply:
I think it would be anyway either one case or the other. 

I think though that it should be safe anyway. If you have a "direct" executor, it is like a dummy executor. 

We could have threads `t1` and `t2` sharing the same instance. when `t1` submits one task, it gets executed immediately in `t1`. `t2` tasks will be executed in `t2` thread.

Reply:
Previously I think it wasn't possible. if getNumJournalCallbackThreads was 0, then we would throw an exception creating the executor. If 1, then callbacks would be serialized. That said, if you had getNumJournalCallbackThreads > 1, you have no guarantee where you run, they would have to be thread safe anyhow.